### ISSUE TITLE
False Positive Style/RedundantFreeze for Array#*

### ISSUE THREAD

**Issue by t-kinoshita** [rubocop-hq/rubocop#9213]
`Array#*` generates false positive `Style/RedundantFreeze`

--------

## Expected behavior

No offence detected

## Actual behavior

(run with `rubocop --debug --only Style/RedundantFreeze`)

```
For /tmp: Default configuration from /home/user/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/rubocop-1.6.1/config/default.yml
Inspecting 1 file
Scanning /tmp/test.rb
C

Offenses:

test.rb:1:7: C: [Correctable] Style/RedundantFreeze: Do not freeze immutable objects, as freezing them has no effect.
FOO = ([0] * 6).freeze
      ^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense auto-correctable
Finished in 0.13516086299932795 seconds
```

## Steps to reproduce the problem

```rb
FOO = ([0] * 6).freeze
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ [bundle exec] rubocop -V
1.6.1 (using Parser 2.7.2.0, rubocop-ast 1.3.0, running on ruby 2.6.6 x86_64-linux)
```
